app/code/Magento/Directory/Model/CurrencySystemConfig.php

```diff
@@ -0,0 +1,123 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+namespace Magento\Directory\Model;
+
+use Magento\Config\App\Config\Type\System;
+use Magento\Framework\App\ResourceConnection;
+use Magento\Store\Model\StoreManagerInterface;
+
+/**
+ * Provide system config values for allowed, base and default currencies.
+ */
+class CurrencySystemConfig
+{
+    /**
+     * @var System
+     */
+    private $systemConfig;
+
+    /**
+     * @var StoreManagerInterface
+     */
+    private $storeManager;
+
+    /**
+     * @var string
+     */
+    private $path;
+
+    /**
+     * Currency constructor.
+     *
+     * @param System $systemConfig
+     * @param StoreManagerInterface $storeManager
+     */
+    public function __construct(
+        System $systemConfig,
+        StoreManagerInterface $storeManager,
+        ResourceConnection $resources
```

Comment:
I guess we do not need this if it is never assigned to anything.

Reply:
Removed.